fix perf test failed

**Self Checks**

To make sure we get to you in time, please check the following :)
- [ ] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify-plugin-daemon/issues), including closed ones.
- [ ] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [ ] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:)
- [ ] "Please do not modify this template :) and fill in all the required fields."

**Versions**
1. dify-plugin-daemon Version
2. dify-api Version

**Describe the bug**
```
Run go test -run=^$ -bench=. -benchmem -benchtime=30s ./...
?   	github.com/langgenius/dify-plugin-daemon/cmd/codegen	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/cmd/commandline	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/cmd/commandline/bundle	[no test files]
PASS
ok  	github.com/langgenius/dify-plugin-daemon/cmd/commandline/plugin	0.005s
?   	github.com/langgenius/dify-plugin-daemon/cmd/commandline/run	[no test files]
PASS
ok  	github.com/langgenius/dify-plugin-daemon/cmd/commandline/signature	0.004s
?   	github.com/langgenius/dify-plugin-daemon/cmd/license/generate	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/cmd/license/sign	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/cmd/license/verify	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/cmd/server	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/cmd/slim	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/cmd/tests	[no test files]
PASS
ok  	github.com/langgenius/dify-plugin-daemon/integration/db	0.007s
PASS
ok  	github.com/langgenius/dify-plugin-daemon/integration/plugin	0.006s
PASS
ok  	github.com/langgenius/dify-plugin-daemon/internal/cluster	0.004s
PASS
ok  	github.com/langgenius/dify-plugin-daemon/internal/core/control_panel	0.006s
2026/04/28 06:34:39 INFO init routine pool size=1024
PASS
ok  	github.com/langgenius/dify-plugin-daemon/internal/core/debugging_runtime	0.015s
?   	github.com/langgenius/dify-plugin-daemon/internal/core/dify_invocation	[no test files]
PASS
ok  	github.com/langgenius/dify-plugin-daemon/internal/core/dify_invocation/calldify	0.005s
?   	github.com/langgenius/dify-plugin-daemon/internal/core/dify_invocation/mock	[no test files]
PASS
ok  	github.com/langgenius/dify-plugin-daemon/internal/core/io_tunnel	0.007s
?   	github.com/langgenius/dify-plugin-daemon/internal/core/io_tunnel/access_types	[no test files]
PASS
ok  	github.com/langgenius/dify-plugin-daemon/internal/core/io_tunnel/backwards_invocation	0.005s
PASS
ok  	github.com/langgenius/dify-plugin-daemon/internal/core/io_tunnel/backwards_invocation/transaction	0.006s
PASS
ok  	github.com/langgenius/dify-plugin-daemon/internal/core/local_runtime	0.005s
redis: 2026/04/28 06:34:40 pool.go:426: redis: connection pool: failed to dial after 5 attempts: dial tcp [::1]:6379: connect: connection refused
redis: 2026/04/28 06:34:40 pool.go:426: redis: connection pool: failed to dial after 5 attempts: dial tcp [::1]:6379: connect: connection refused
redis: 2026/04/28 06:34:41 pool.go:426: redis: connection pool: failed to dial after 5 attempts: dial tcp [::1]:6379: connect: connection refused
redis: 2026/04/28 06:34:41 pool.go:426: redis: connection pool: failed to dial after 5 attempts: dial tcp [::1]:6379: connect: connection refused
panic: Failed to init redis client: dial tcp [::1]:6379: connect: connection refused

	/home/runner/work/dify-plugin-daemon/dify-plugin-daemon/pkg/utils/log/log.go:113
github.com/langgenius/dify-plugin-daemon/internal/db.Init(0xc00039f780)
	/home/runner/work/dify-plugin-daemon/dify-plugin-daemon/internal/db/init.go:110 +0x64b
github.com/langgenius/dify-plugin-daemon/internal/types/models/curd.TestMain(0xc000323860)
	/home/runner/work/dify-plugin-daemon/dify-plugin-daemon/internal/types/models/curd/testmain_test.go:24 +0x105
main.main()
	_testmain.go:53 +0xa5
exit status 2
FAIL	github.com/langgenius/dify-plugin-daemon/internal/types/models/curd	0.006s
PASS
ok  	github.com/langgenius/dify-plugin-daemon/pkg/bundle_packager	0.003s
?   	github.com/langgenius/dify-plugin-daemon/pkg/entities	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/pkg/entities/agent_entities	[no test files]
PASS
ok  	github.com/langgenius/dify-plugin-daemon/pkg/entities/bundle_entities	0.003s
?   	github.com/langgenius/dify-plugin-daemon/pkg/entities/constants	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/pkg/entities/datasource_entities	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/pkg/entities/dynamic_select_entities	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/pkg/entities/endpoint_entities	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/pkg/entities/installation_entities	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/pkg/entities/manifest_entities	[no test files]
PASS
ok  	github.com/langgenius/dify-plugin-daemon/pkg/entities/model_entities	0.003s
?   	github.com/langgenius/dify-plugin-daemon/pkg/entities/oauth_entities	[no test files]
PASS
ok  	github.com/langgenius/dify-plugin-daemon/pkg/entities/plugin_entities	0.003s
?   	github.com/langgenius/dify-plugin-daemon/pkg/entities/requests	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/pkg/entities/tool_entities	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/pkg/license/private_key	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/pkg/license/public_key	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/pkg/manifest	[no test files]
PASS
ok  	github.com/langgenius/dify-plugin-daemon/pkg/plugin_packager	0.003s
?   	github.com/langgenius/dify-plugin-daemon/pkg/plugin_packager/consts	[no test files]
PASS
ok  	github.com/langgenius/dify-plugin-daemon/pkg/plugin_packager/decoder	0.003s
?   	github.com/langgenius/dify-plugin-daemon/pkg/plugin_packager/packager	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/pkg/plugin_packager/signer	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/pkg/plugin_packager/signer/withkey	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/pkg/routine	[no test files]
PASS
ok  	github.com/langgenius/dify-plugin-daemon/pkg/slim	0.003s
PASS
ok  	github.com/langgenius/dify-plugin-daemon/pkg/utils/cache	0.003s
?   	github.com/langgenius/dify-plugin-daemon/pkg/utils/cache/helper	[no test files]
PASS
ok  	github.com/langgenius/dify-plugin-daemon/pkg/utils/debugging	0.001s
?   	github.com/langgenius/dify-plugin-daemon/pkg/utils/encryption	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/pkg/utils/http_requests	[no test files]
PASS
ok  	github.com/langgenius/dify-plugin-daemon/pkg/utils/json_schema	0.002s
PASS
ok  	github.com/langgenius/dify-plugin-daemon/pkg/utils/lock	0.001s
PASS
ok  	github.com/langgenius/dify-plugin-daemon/pkg/utils/log	0.002s
PASS
ok  	github.com/langgenius/dify-plugin-daemon/pkg/utils/mapping	0.001s
?   	github.com/langgenius/dify-plugin-daemon/pkg/utils/metrics	[no test files]
PASS
ok  	github.com/langgenius/dify-plugin-daemon/pkg/utils/network	0.002s
PASS
ok  	github.com/langgenius/dify-plugin-daemon/pkg/utils/parser	0.002s
?   	github.com/langgenius/dify-plugin-daemon/pkg/utils/routine	[no test files]
goos: linux
goarch: amd64
pkg: github.com/langgenius/dify-plugin-daemon/pkg/utils/stream
cpu: AMD EPYC 9R45
BenchmarkStream-2   	1000000000	        20.09 ns/op	      15 B/op	       0 allocs/op
PASS
ok  	github.com/langgenius/dify-plugin-daemon/pkg/utils/stream	22.131s
?   	github.com/langgenius/dify-plugin-daemon/pkg/utils/strings	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/pkg/utils/tmpfile	[no test files]
?   	github.com/langgenius/dify-plugin-daemon/pkg/validators	[no test files]
FAIL
```

**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

--- a/internal/core/persistence/persistence_test.go
+++ b/internal/core/persistence/persistence_test.go
@@ -2,24 +2,36 @@ package persistence
 
 import (
 	"encoding/hex"
+	"fmt"
 	"os"
 	"testing"
+	"time"
 
 	cloudoss "github.com/langgenius/dify-cloud-kit/oss"
 	"github.com/langgenius/dify-cloud-kit/oss/factory"
 	"github.com/langgenius/dify-plugin-daemon/internal/db"
 	"github.com/langgenius/dify-plugin-daemon/internal/types/app"
 	"github.com/langgenius/dify-plugin-daemon/pkg/utils/cache"
+	"github.com/langgenius/dify-plugin-daemon/pkg/utils/network"
 	"github.com/langgenius/dify-plugin-daemon/pkg/utils/strings"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestMain(m *testing.M) {
+	if !network.IsTCPReachable("localhost:6379", 500*time.Millisecond) {
+		fmt.Fprintln(os.Stderr, "skipping persistence tests: redis is unavailable at localhost:6379")
+		os.Exit(0)
+	}
 	if err := cache.InitRedisClient("localhost:6379", "", "difyai123456", false, 0, nil); err != nil {
-		panic("Failed to init redis client: " + err.Error())
+		fmt.Fprintf(os.Stderr, "skipping persistence tests: failed to init redis client: %v\n", err)
+		os.Exit(0)
 	}
 	defer cache.Close()
 
+	if !network.IsTCPReachable("localhost:5432", 500*time.Millisecond) {
+		fmt.Fprintln(os.Stderr, "skipping persistence tests: postgres is unavailable at localhost:5432")
+		os.Exit(0)
+	}
 	db.Init(&app.Config{
 		DBType:     app.DB_TYPE_POSTGRESQL,
 		DBUsername: "postgres",

--- a/internal/types/models/curd/testmain_test.go
+++ b/internal/types/models/curd/testmain_test.go
@@ -1,15 +1,23 @@
 package curd
 
 import (
+	"fmt"
 	"os"
 	"testing"
+	"time"
 
 	"github.com/langgenius/dify-plugin-daemon/internal/db"
 	"github.com/langgenius/dify-plugin-daemon/internal/types/app"
 	"github.com/langgenius/dify-plugin-daemon/pkg/utils/cache"
+	"github.com/langgenius/dify-plugin-daemon/pkg/utils/network"
 )
 
 func TestMain(m *testing.M) {
+	if !network.IsTCPReachable("localhost:5432", 500*time.Millisecond) {
+		fmt.Fprintln(os.Stderr, "skipping curd tests: postgres is unavailable at localhost:5432")
+		os.Exit(0)
+	}
+
 	cfg := &app.Config{
 		DBType:     app.DB_TYPE_POSTGRESQL,
 		DBUsername: "postgres",
@@ -28,8 +36,14 @@ func TestMain(m *testing.M) {
 		panic("Failed to auto migrate: " + err.Error())
 	}
 
+	if !network.IsTCPReachable("127.0.0.1:6379", 500*time.Millisecond) {
+		fmt.Fprintln(os.Stderr, "skipping curd tests: redis is unavailable at 127.0.0.1:6379")
+		os.Exit(0)
+	}
+
 	if err := cache.InitRedisClient("127.0.0.1:6379", "", "difyai123456", false, 0, nil); err != nil {
-		panic("Failed to init redis client: " + err.Error())
+		fmt.Fprintf(os.Stderr, "skipping curd tests: failed to init redis client: %v\n", err)
+		os.Exit(0)
 	}
 	defer cache.Close()
 

--- a/pkg/utils/network/reachability.go
+++ b/pkg/utils/network/reachability.go
@@ -1,0 +1,15 @@
+package network
+
+import (
+	"net"
+	"time"
+)
+
+func IsTCPReachable(addr string, timeout time.Duration) bool {
+	conn, err := net.DialTimeout("tcp", addr, timeout)
+	if err != nil {
+		return false
+	}
+	_ = conn.Close()
+	return true
+}

